lib/webdriveragent-utils.js

```diff
@@ -0,0 +1,60 @@
+import { fs } from 'appium-support';
+import replaceInFile from 'replace-in-file';
+import { exec } from 'teen_process';
+import log from './logger';
+
+
+const PROJECT_FILE = 'project.pbxproj';
+
+async function updateProjectFile (agentPath, oldBundleId, newBundleId) {
+  let projectFilePath = `${agentPath}/${PROJECT_FILE}`;
+  try {
+    // backup the file, and then update the bundle id for the runner
+    await fs.copyFile(projectFilePath, `${projectFilePath}.old`);
+    let changedFiles = await replaceInFile({
+      files: projectFilePath,
+      replace: oldBundleId,
+      with: newBundleId,
+    });
+    log.debug(`Successfully updated '${changedFiles.join(', ')}' with bundle id '${newBundleId}'`);
+  } catch (err) {
+    log.debug(`Error updating project file: ${err.message}`);
+    log.warn(`Unable to update project file '${projectFilePath}' with ` +
+             `bundle id '${newBundleId}'. WebDriverAgent may not start`);
+  }
+}
+
+async function resetProjectFile () {
```

Comment:
interesting. I think it's probably worth pursuing the reset logic. When in doubt, leaving things the way we found them is probably the path of least surprise.